packages/web-ui/src/ds/molecules/MultiSelect/index.tsx

```diff
@@ -0,0 +1,362 @@
+// src/components/multi-select.tsx
+
+import * as React from 'react'
+
+import { cn } from '../../../lib/utils'
+import {
+  Badge,
+  Checkbox,
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  FormField,
+  Popover,
+  Separator,
+} from '../../..'
+import { Button } from '../../ui/button'
+import Text from '../../atoms/Text'
+import { Icon } from '../../atoms/Icons'
+
+/**
+ * Props for MultiSelect component
+ */
+interface MultiSelectProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
+    Omit<typeof FormField, 'children'> {
+  /**
+   * An array of option objects to be displayed in the multi-select component.
+   * Each option object has a label, value, and an optional icon.
+   */
+  options: {
+    /** The text to display for the option. */
+    label: string
+    /** The unique value associated with the option. */
+    value: string
+    /** Optional icon component to display alongside the option. */
+    icon?: React.ComponentType<{ className?: string }>
+  }[]
+
+  /**
+   * Callback function triggered when the selected values change.
+   * Receives an array of the new selected values.
+   */
+  onChange?: (value: string[]) => void
+
+  /** The default selected values when the component mounts. */
+  defaultValue?: string[]
+
+  /**
+   * Placeholder text to be displayed when no values are selected.
+   * Optional, defaults to "Select options".
+   */
+  placeholder?: string
+
+  /**
+   * Animation duration in seconds for the visual effects (e.g., bouncing badges).
+   * Optional, defaults to 0 (no animation).
+   */
+  animation?: number
+
+  /**
+   * Maximum number of items to display. Extra selected items will be summarized.
+   * Optional, defaults to 3.
+   */
+  maxCount?: number
+
+  /**
+   * The modality of the popover. When set to true, interaction with outside elements
+   * will be disabled and only popover content will be visible to screen readers.
+   * Optional, defaults to false.
+   */
+  modalPopover?: boolean
+
+  /**
+   * If true, renders the multi-select component as a child of another component.
+   * Optional, defaults to false.
+   */
+  asChild?: boolean
+
+  /**
+   * Additional class names to apply custom styles to the multi-select component.
+   * Optional, can be used to add custom styles.
+   */
+  className?: string
+
+  /**
+   * Additional description to display below the multi-select component.
+   * Optional, can be used to provide additional information or instructions.
+   */
+  description?: string
+
+  /**
+   * Additional information to display in a tooltip over the multi-select component.
+   * Optional, can be used to provide additional information or instructions.
+   */
+  info?: string
+
+  /**
+   * Additional label to display above the multi-select component.
+   */
+  label?: string
```

Comment:
Do we need all these comments? too much no?

Reply:
yeah no